### ISSUE TITLE
analog mezzanine physical/interface specification

### ISSUE THREAD

**Issue by jordens** [m-labs/sinara#8]
Here is the final summary, based on comments in the thread below.  This post will be sticky at the top, containing the most current specification.  Items labeled **_ACTION:_** are not finalized yet; @dhslichter will edit this comment with the correct information as it is finalized.

## Mezzanine geometry/stackup
- mezzanine board 32 x 100 mm, .047" nominal thickness, 4 layer boards
    - top and bottom layers to be Rogers 4350B, .020" thickness (cap construction for PCBs)
    - 50 ohm CPWG on .020" Rogers 4350B is .035" trace with .011" gaps
    - need ground stitch vias every ~.050" or less along both sides of CPWG
- mezzanines are mounted to RTM motherboard using 4x M1.5 screws
- 9 mm stand off between motherboard and mezzanine
- RF shields on both sides, separate cavities for each DAC channel and ADC channel (4 total per mezzanine)
    - interior height under DAC side shield lids needs to be .170" minimum (allowing for fabrication tolerances) to accommodate baluns
    - interior height under ADC side shield lids .120" (3 mm).  

## Connectors

- power/IO:  Samtec hermaphroditic, 40-pins connectors: LSS-120-01-L-DV-A and LSS-120-02-L-DV-A
- RF to/from motherboard
    - DACs: 4 SMP (2 differential pairs) - Molex 85305-0232
    - ADCs: 4 SMP (2 differential pairs) - SMP-MSLD-PCT-10 
    - clock: 2 SMP (1 differential pair) generated on sayma_clk mezzanine - either style, depending on side of board where clock circuitry is needed.
- panel RF connection
    - see #7 
    - option to have SMA shield isolated from Sayma ground (i.e. not grounded to panel) -- use plastic insulating washers on both sides of panel (e.g. http://www.rf-microwave.com/datasheets/603_Huber-Suhner_77-Z-0-0-8_01.pdf)
    - DAC side of board: 2 SMA (Telegartner J01151A0451)
    - ADC side of board: 2 SMA (Telegartner J01151A0921)
    - _**ACTION:**_ use multi-coax assembly for front panel?

## Power rails

- +12VDC @ 200 mA, max 1 mV p-p noise in 20 Hz-20 MHz bandwidth
- +6VDC @ 1.5 A, max 1 mV p-p noise in 20 Hz-20 MHz bandwidth
- -12VDC @ 50 mA, max 1 mV p-p noise in 20 Hz-20 MHz bandwidth
- -6VDC @ 150 mA, max 1 mV p-p noise in 20 Hz-20 MHz bandwidth
- no switching "spikes" on rails
- pi filters on all power lines
    - 0402 NP0 1 nF cap (16V min rating) parallel 0805 X7R 10 uF cap (16V min rating - Samsung CL21B106KOQNNN or equivalent DC bias performance -- degrades to 7.5 uF @ 3.3V, 5.5 uF @ 6V, 2.7 uF @ 12V) to ground, between mezzanine and choke
    - BLM41PG181SN1L choke
    - 0805 X7R 10 uF cap to ground (16V min rating - Samsung CL21B106KOQNNN or equivalent DC bias performance), between choke and supply
- desired supply voltages to be generated from these rails on mezzanine with dedicated LDOs
- no switching supplies on mezzanines
- +3.3VDC @ 1 A for digital-only devices on mezzanine.  Should still be filtered on Sayma to minimize noise. max 10 mV p-p noise in 20 Hz-20 MHz bandwidth acceptable.
- single shared ground for analog and digital, connected to all SMP shields and all ground pins of power/IO connector

## Power/IO header routing

- as shown in image below from @gkasprow with exceptions listed below
- power line filters to pi filters as described above.  
- digital lines IO0-IO15 3.3V LVCMOS
- I2C SDA/SCL (3.3V), pullup resistors to +3.3V installed on Sayma.  I2C fed from I2C mux chips on Sayma card.  Easy/compact level shifting on mezzanines (if needed) using discrete MOSFETs: http://www.nxp.com/documents/application_note/AN10441.pdf
- digital IO are general purpose in hardware, but for SPI preferred pinout is:
    - SCLK = IO0
    - MOSI = IO1
    - MISO = IO2
    - CS = IO3
    - additional chip selects starting with IO4 and up
- power rails as shown (7V rail is actually 6V)
- 4x connections to slow ADC (24-bit sigma-delta) for drift/temp/etc monitoring.  ADC chip is mounted on  See #11.

![image](https://cloud.githubusercontent.com/assets/9222487/19248583/342c2786-8eee-11e6-9b3d-df31b5e45433.png)

**Comment by gkasprow** [comment]
I did some estimation.
We can easily fit four 32x100mm DAC/ADC  mezzanines on RTM. We can even make them longer.
On each of them we will fit four Wurth 30x30mm shields placed on both sides.
In case of RF connectors I consider using:
2 high performance SMA, edge mount for DAC output
2 lower performance, right angled for ADC input.
8 SMP connectors for ADC/DAC. THT version with SMD central pad would give best mechanical properties. The distance between mezzanine and RTM is 9mm which is minimal value for SMP+bullet.
For digital IO and control I'd like to use Samtec hermaphroditic, 40 pins connectors: LSS-120-01-L-DV-A and LSS-120-02-L-DV-A  which combined give exactly 9mm. I use them in other projects and they work well with frequent mating.
Mechanical mounting would be ensured by 4 x 2.6mm holes, 4 threaded spacers and 8 M2.5 screws.
There is still place on front panel for SMA clock input 
I placed typical bulky SMA connectors to check the clearance between them. It is not that bad. Then are accessible with the wrench when the RTM is outside of the crate.
Attached are PDF3D files with mezzanine and RTM.
[mezzanine.pdf](https://github.com/m-labs/sayma/files/501839/mezzanine.pdf)
[RTM_DAC.pdf](https://github.com/m-labs/sayma/files/501840/RTM_DAC.pdf)
We will leave the space below mezzanines to enable the air flow. Some cut-outs in RTM may be necessary as well.

**Comment by gkasprow** [comment]
The clock module dimension is 43x80mm
We can make it slightly longer when necessary
here is the RTM with clock module
[RTM_DAC_clk.pdf](https://github.com/m-labs/sayma/files/504373/RTM_DAC_clk.pdf)

**Comment by jordens** [comment]
Thanks! Impressive.

* Is this still with the SMA firmly screwed into the front panel? How would the mezzanines mounted in that case?
* Did we want no common LO clock (for whatever upconversion needs there may be) to the mezzanines?
* I assume that there is still space for the dacs, adcs, power supplies, and clock fanout?

**Comment by sbourdeauducq** [comment]
I cannot open any of those PDFs, either with mupdf or Evince.
```
$ mupdf mezzanine.pdf 
error: expected trailer marker
error: cannot parse trailer
error: cannot read xref (ofs=617742)
error: cannot read xref at offset 617742
```

**Comment by whitequark** [comment]
@sbourdeauducq Yes. As far as I know PDF3D can be opened only with proprietary Adobe Reader for Linux.

**Comment by sbourdeauducq** [comment]
Adobe Reader freezes when trying to open those documents.

**Comment by sbourdeauducq** [comment]
Is there another format we can use for this?

**Comment by gkasprow** [comment]
we can use step


On 1 October 2016 at 18:10, Sébastien Bourdeauducq <notifications@github.com

**Comment by gkasprow** [comment]
To view step files I use free Design Spark Mechanic.

**Comment by whitequark** [comment]
FreeCAD is very bad but it should suffice to view STEP files.

**Comment by gkasprow** [comment]
Below are step files. Due to 10MB limit I zipped them

**Comment by gkasprow** [comment]
Due to 10MB limit I had to zip them
[CLK module.zip](https://github.com/m-labs/sayma/files/504635/CLK.module.zip)
[mezzanine2.zip](https://github.com/m-labs/sayma/files/504636/mezzanine2.zip)
[RTM_DAC_clk.zip](https://github.com/m-labs/sayma/files/504637/RTM_DAC_clk.zip)

**Comment by gkasprow** [comment]
In case it pdf 3D files you have to use recent Adobe Acrobat Reder

**Comment by gkasprow** [comment]
> Is this still with the SMA firmly screwed into the front panel? How would the mezzanines mounted in >that case?

the mezzanines are mounted to the RTM board using 4 M1.5 screws 

>    Did we want no common LO clock (for whatever upconversion needs there may be) to the >mezzanines?
I'll add another SMP connector
   > I assume that there is still space for the dacs, adcs, power supplies, and clock fanout?
sure, some of them can be plalced below the mezzanines

**Comment by gkasprow** [comment]
>    Is this still with the SMA firmly screwed into the front panel?

There are versions with long thread both right angle and straddle mounted.
But their performance need to be verified since they are low cost.
http://www.amphenolrf.com/132289.html
http://www.rfstreet.com/ch_contentd.asp?id=2043
on the other side the assembly of the modules changed - they will be firmly attached to the rigid RTM board by 4 screws so it may be not necessary to screw them to the front panel

**Comment by sbourdeauducq** [comment]
Thanks. The STEP files work fine with FreeCAD. FYI, the Acrobat Android app and the Chrome PDF viewer also choke on the PDF3Ds.

**Comment by sbourdeauducq** [comment]
What is the idea behind having more SMPs than SMAs on the mezzanines? Since this also results in more SMPs than DAC/ADC channels on the RTM carrier, how will they be connected?

**Comment by gkasprow** [comment]
Did you enable the 3D view in AR?

It needs a while while opening the file

Greg

**Comment by gkasprow** [comment]
The signals of the ADCs and DACs are differential, that's why we need the SMPs doubled.
The mezzanines inputs and outputs are single ended.
The idea is to place baluns on the mezzanines because the way one feeds the ADC or couples to DAC depends on application.

**Comment by sbourdeauducq** [comment]
Aren't SMPs designed for single-ended unbalanced signals?

**Comment by gkasprow** [comment]
Well, in this way you can say that traces on PCB are designed for single
ended signals.
If you make differential traces that are loosely coupled, you can treat
them as single ended. When you make traces tightly coupled, then they
interfere with each other and if you split them then their impedance will
increase.
The same applies to SMAs - on every devkit you can connect gigabit
transceivers using single-ended connectors.
To transfer signals using SMPs you have to make impedance conversion from
tightly coupled diff line to loosely coupled diff line and then transfer
via connectors.
So to make 100Ohm diff line you can use 2 traces wit 50Ohm but separated
far away or use two 70Ohm traces coupled in such way that trace field see
each other. In first case you can transfer them wia SMP, SMA or whatever,
in second case you have to transfer the line to the first case.


On 2 October 2016 at 17:08, Sébastien Bourdeauducq <notifications@github.com

**Comment by jordens** [comment]
This works perfectly fine. Convince yourself using the method of image charges and symmetries.

**Comment by dhslichter** [comment]
It's very hard to find true differential signal connectors where the channel-to-channel crosstalk is anywhere near as low as if one uses two coaxial connectors (e.g. SMP) for a differential signal.  Sending the ADC/DAC differential signals over pairs of SMP connectors is the best compromise we could come up with given crosstalk/space/price/bandwidth/mating limitations and requirements.

**Comment by jbqubit** [comment]
Impressive drawings! 
> Wurth 30x30mm shields placed on both sides
- Should these also cover the back side where the SMP penetrate? 
- A single shield on each side would give more freedom for component placement. 
- What's the maximum vertical height for components? Is this sufficient for bulky components like baluns, transformers?

**Comment by dhslichter** [comment]
Several points:

- if board will have components on both sides, should use Rogers 4350B on both sides.  OK to make it thinner than .020" -- 50 ohm CPWG on .020" Rogers 4350B is 35 mil trace with 11 mil gaps, based on measurements at NIST.
- shields should be multi-cavity (if a single part) to reduce crosstalk between channels on the same daughtercard.  Something along the lines of https://leadertechinc.com/product/multi-cavity-slot-lok/ would be good.
- interior height under shield lids needs to be .165" minimum to accommodate baluns
- crosstalk reduction would be improved if shields extend all the way to the end of the board, such that SMP connectors are connected through holes in the shields.  Unshielded lengths should be short to the SMA connectors.  Notches need to be made in the shields to avoid impedance discontinuities where CPWG traces exit from under the shields to the SMA connectors.

**Comment by gkasprow** [comment]
The Wurth shields were only placed just to reserve place on PCB. There are
smaller ones like these
http://www.digikey.com/product-search/en/rf-if-and-rfid/rf-shields/3539677?k=&pkeyword=&pv1217=1&FV=fff40036%2Cfff802dd%2C13040001%2C13040009&mnonly=0&newproducts=0&ColumnSort=0&page=1&quantity=0&ptm=0&fid=0&pageSize=25

**Comment by dhslichter** [comment]
Again, the shields you quote are not tall enough inside, nor would they extend the full size of the board.  I understand the Wurth shields are placeholders, but I think we will probably want something custom/multicavity that covers as much of the board as possible while separating the various channels from each other.

**Comment by gkasprow** [comment]
In this case we can use custom Harwin shields (EZ-shield) where they can be
freely adjusted with 5mm step.
You just install clips on the PCB and then manually form the cover.
http://www.mouser.com/new/harwin/harwin-ez-emi-rfi-shield-cans/

**Comment by jordens** [comment]
OK. @dhslichter and @gkasprow if you feel that you have arrived at a workable solution, please close the issue. AFAICT the rest has weighed in.

**Comment by dhslichter** [comment]
@gkasprow can you clarify on which SMA connectors you want to use on the front panel, and their layout?  Let's try to put the whole spec into a single comment at the end of the thread before we close.

**Comment by jbqubit** [comment]
Here's my summary of current design choices for mezzanine interface (and geometry) for each of 4 analog boards. @jordens @gkasprow @sbourdeauducq @dhslichter please review. @weidazh please forward to Harty

**geometry and mechanical**
- what is mezzanine geometry?
- the mezzanines are mounted to RTM motherboard using 4 of M1.5 screws

**RF shield**
- interior height under shield lids needs to be .165" minimum to accommodate baluns

**PCB material/layers**
- 4-layer PCB, top and bottom layers .020" Rogers 4350B
    - 50 ohm CPWG on .020" Rogers 4350B is 35 mil trace with 11 mil gaps
- single ground for analog and digital

**connector choices**
- power and IO
  - Samtec hermaphroditic, 40-pins connectors: LSS-120-01-L-DV-A and LSS-120-02-L-DV-A
  - 9 mm stand off between motherboard and mezzanine 
  - what is maximum current?
- RF to/from motherboard
    - analog output: 4 SMP (2 differential pairs)
    - analog input: 4 SMP (2 differential pairs) 
    - clock: 2 SMP (1 differential pair) generated on sayma_clk mezzanine 
    - what is part number for SMP's? Discussed in #7 but not yet finalized? 
- RF to outside world
    - SMA choice discussed in #7 but not yet finalized? 

**power/IO header routing**
- digital
     - 18 digital lines total 
     - no pull-up resistors
     - +3.3V LVCMOS single-ended to/from FPGA
     - recommended use:
         - pin names in schematic should include names indicative of recommended use case
         - I2C [I2C_SDA, I2C_SCL] 
         - SPI [SPI_SCLK, SPI_MOSI, SPI_MISO, SPI_CS0, SPI_CS1, SPI_CS2, SPI_CS3, ...]

- analog power, low noise (generated on AMC)
    - +12V at 200 mA

Does github permit making an Issue comment as "sticky" so it rises to top? Seems like that's where the summary should lie. 
    - -12V at 200 mA
    - +7V at 1 A
    - -7V at 1 A
    - +3.3V at 1 A
    - other potentials generated on-mezzanine using low-dropout regulators (LDR) -- no switching regulators on mezzanine
    - schematic notation to distinguish between analog and digital power
- digital power
    - +3.3 V at 1 A

**ADCs**
- 4 channels

**Comment by jbqubit** [comment]
@jordens Does github permit making an Issue comment as "sticky" so it rises to top? Seems like that's where the summary should lie.

**Comment by gkasprow** [comment]
During our last discussion we decided:
+6V 1.5A. This minimizez LDO losses. We don't need -7 for RF stuff.
+12V 200mA
-12V 50mA
no need for other power lines.
I'd add local I2C line for module identification - the same mechanism as is on the FMCs.
In this way we will quickly know which modules are installed in certain mezzanines. With all system enclosed in the rack it's good to know what is inserted in each RTM.
To save pins, the I2C lines can be connected to the I2C muxes and later on to the Artix FPGA and in parallel to the IPMI link. To supply EEPROM and some digital stuff 3.3V is needed with low current capability.
Now the connector looks like this;
![mezzanine connector1](https://cloud.githubusercontent.com/assets/4325054/19245996/62d5c21e-8f24-11e6-9e5a-a19a1bf79992.PNG)

**Comment by dhslichter** [comment]
Here is the final summary, based on comments from @gkasprow  and @jbqubit in this thread.  Items labeled **_ACTION:_** are not finalized yet; I will edit this comment with the correct information once they are finalized.  @jordens @gkasprow @sbourdeauducq @jbqubit @weidazh please provide feedback.

## Mezzanine geometry/stackup
- mezzanine board 32 x 100 mm, .062" nominal thickness, 4 layer boards
    - top and bottom layers to be Rogers 4350B, .020" thickness (cap construction for PCBs)
    - 50 ohm CPWG on .020" Rogers 4350B is .035" trace with .011" gaps
    - need ground stitch vias every ~.050" or less along both sides of CPWG
- mezzanines are mounted to RTM motherboard using 4x M1.5 screws
- 9 mm stand off between motherboard and mezzanine
- RF shields on both sides, separate cavities for each DAC channel and ADC channel (4 total per mezzanine)
    - interior height under shield lids needs to be .170" minimum (allowing for fabrication tolerances) to accommodate baluns

## Connectors

- power/IO:  Samtec hermaphroditic, 40-pins connectors: LSS-120-01-L-DV-A and LSS-120-02-L-DV-A
- RF to/from motherboard
    - DACs: 4 SMP (2 differential pairs)
    - ADCs: 4 SMP (2 differential pairs)
    - clock: 2 SMP (1 differential pair) generated on sayma_clk mezzanine
    - **_ACTION:_** need to finalize SMP part number
- panel RF connection
    - option to have SMA shield isolated from Sayma ground (i.e. not grounded to panel)
    - **_ACTION:_** need to finalize SMA part numbers

## Power rails

- +12VDC @ 200 mA, low noise 
- +6VDC @ 1.5 A, low noise
- -12VDC @ 50 mA, low noise
- **_ACTION:_** need to specify noise requirement for above rails
- desired supply voltages to be generated from these rails on mezzanine with dedicated LDOs
- no switching supplies on mezzanines
- **_ACTION:_** +3.3VDC @ 1 A for digital-only devices on mezzanine?
- single shared ground for analog and digital, connected to all SMP shields and all ground pins of power/IO connector


## Power/IO header routing

- as shown in image below from @gkasprow 
- digital lines IO0-IO15 3.3V LVCMOS
- **_ACTION:_** need to choose I2C SDA/SCL voltage, and whether pullups are on Sayma or mezzanine (recommend Sayma).  Easy/compact level shifting on mezzanines (if needed) using discrete MOSFETs: http://www.nxp.com/documents/application_note/AN10441.pdf
- digital IO are general purpose in hardware, but for SPI preferred pinout is:
    - SCLK = IO0
    - MOSI = IO1
    - MISO = IO2
    - CS = IO3
    - additional chip selects starting with IO4 and up
- power rails as shown (7V rail is actually 6V)
- 4x connections to slow ADC for drift/temp/etc monitoring. **_ACTION:_** confirm ADC to live on Sayma board, not on mezzanine.

![image](https://cloud.githubusercontent.com/assets/9222487/19248583/342c2786-8eee-11e6-9b3d-df31b5e45433.png)

**Comment by gkasprow** [comment]
I2C voltage is 3.3V.
The calibration ADC is on the RTM board, controlled by Artix FPGA>

**Comment by gkasprow** [comment]
SMP part no is: 
MOLEX 85305-0232 for DAC signals
Amphenol SMP-MSLD-PCT-10 for ADC signals 
With such scenario we will keep DAC signals on same layer and ADC signals on opposite layer.
SMA part no is:
Telegartner J01151A0451 for DAC signals. It is the only choice for desired  performance.
Telegartner J01151A0921 for ADC signals. This selection is forced by type of SMAs used for DACs. No other type of connector will fit.

**Comment by dhslichter** [comment]
There are still 4 unused pins on the power/IO header.  Would it make sense to use these to make the ADC input lines into differential pairs?  Thus there would be 8 pins total going to the slow ADC, which could be either used as 4 differential pairs or 8 single-ended lines with a chip like http://www.analog.com/en/products/analog-to-digital-converters/ad-converters/ad7173-8.html

**Comment by gkasprow** [comment]
I'd reserve them for future use.
For sure we will find some application for them

**Comment by dhslichter** [comment]
@gkasprow especially on the +12V and +6V rails, we need to make sure that the power line filter components are carefully chosen.  0201 package 1 nF capacitor will likely have very substantial DC bias dependence to capacitance (esp. if caps are only 16V rated, I would use larger margin for +12V rail), for example, and we should check choke impedance at 1.5A load for +6V rail.

**Comment by dhslichter** [comment]
Temperature monitoring can also be performed on mezzanines by using [ADT7420](http://www.analog.com/en/products/analog-to-digital-converters/integrated-special-purpose-converters/integrated-temperature-sensors/adt7420.html), which contains conversion and I2C hardware, thus freeing slow ADC lines for e.g. monitoring of I/Q modulator drift or amplifier gain drift.

**Comment by gkasprow** [comment]
we can use 0402 NP0 instead

**Comment by gkasprow** [comment]
The chokes are rated for 3.5A BLM41PG181SN1L

**Comment by dhslichter** [comment]
I vote 0402 NP0 for the 1 nF, and X7R with higher voltage rating on the 10 uF if possible.  Chokes look good.

**Comment by sbourdeauducq** [comment]
> thus freeing slow ADC lines for e.g. monitoring of I/Q modulator drift or amplifier gain drift.

We also have an ADC inside the XC7A15T.

**Comment by jordens** [comment]
@jbqubit rises to the top of what? What you perceive as important might not be what is currently relevant to others. If you are certain this particular issue is more important than the other "important" issues, give it the aproriate label. You have lots of options to view issues in the way you prefer.
* make yourself a [project dashboard](https://github.com/m-labs/sayma/projects) where you can organize issues for yourself
* add [labels](https://github.com/m-labs/sayma/labels) and 
* create permanent [search links](https://help.github.com/articles/searching-issues/) with special [sorting](https://help.github.com/articles/sorting-issues-and-pull-requests/) to [stuff with certain labels](https://github.com/m-labs/sayma/issues?q=is%3Aissue+is%3Aopen+label%3Aprio%3A2-important) and
* bookmark them or add them to a [wiki page](https://github.com/m-labs/sayma/wiki) or [share them](https://help.github.com/articles/sharing-filters/) by other means.

@dhslichter looks good to me (apart from my usual concerns about layout density)
* Pi-filter (add caps after the chokes) so that the loads don't look into the high impedance chokes?
* I'd either keep editing the top post so that the status quo is immediately visible, or - if history is desired - edit a wiki page or text file in the repository for this, or split this into smaller issues.

**Comment by dhslichter** [comment]
@sbourdeauducq I am less concerned about the number of available ADC channels, was more thinking about the number of ADC signal lines coming out of the mezzanine (set at 4 right now).  For a typical IQ modulator setup, we might be monitoring voltages (via lowpass filters) at the I port, Q port, RF port, and after the output amp, thus would use all the ADC lines in the I/O header for those and not have any for temp monitoring; the I2C temp monitor helps with that.

**Comment by dhslichter** [comment]
@jordens ack.

**Comment by dhslichter** [comment]
@gkasprow where will the SMP connectors providing a 100 MHz clock to the mezzanines go?

**Comment by gkasprow** [comment]
Do you mean RF_LO signal?
The idea is to provide RF clock for up/down converters on the mezzanines.
So I want to place SMP connectors routed to the clock module which may in the future provide such signal
In case of the EXT_100M_RF_IN, it is routed directly to the SMA on RTM front panel.

**Comment by dhslichter** [comment]
I am thinking of a reference clock (e.g. 100 MHz) that will allow one to place a VCO/PLL on the mezzanine for generation of an LO to be mixed with the I/Q signals from the DACs to give an upconverted waveform.  We had discussed the option of this architecture in addition to having the LO generated off-card and sent to the mezzanines via front-panel SMA.  

So yes, one would want to have an SMP pair routed to the clock module from the Sayma board, or else via an SMA cable from the clock mezzanine.  I believe Tom Harty had some notions on this laid out.

**Comment by hartytp** [comment]
@dhslichter @gkasprow I've attached a sketch showing the way I assumed we'd generate + distribute the DAC clock + LO. I've missed some of the recent discussions, so apologies if this is hopelessly out of data/no longer applicable!

Comments:

1.**I'd prefer not to generate the LO on the main clock mezzanine**: since the range of DAC clocks we need is fairly narrow (say, 500MHz - 2.4GHz), we can make a single DAC clock PLL design, with VCO + loop-filer population options to set the frequency. In contrast, the range of LOs spans ~1GHz to >10GHz. The PLL one uses to generate a 12GHz LO for Ytterbium is likely to look very different to, say, a 3GHz LO for calcium. e.g. the Ytterbium PLL might be a 2-loop design, with a 100MHz=>1GHz loop, followed by a 1GHz=>12GHz loop, while the calcium can probably be done as a population option for the DAC clock board. Separating these jobs onto different mezzanines allows one to use the same DAC clock board, regardless of what LO one needs, which seems nice.

2.**We should route the 100MHz reference clock to the analog mezzanine**: for LO frequencies up to a few GHz, one can buy IQ modulators with integrated PLL + VCO to generate their own LO from the 100MHz reference clock (e.g. HMC1197). These provide a very convenient, low-noise method of upconversion.

3.**We should have the option of installing a LO generation mezzanine (separate to the DAC clock mezzanine)**: one can put an ultra-low noise PLL on this board to generate the LO from the reference clock. If this is done on the output mezzanines then it has to be duplicated 4 times per Sayma board, which wastes space + money. This is particularly important for LO frequencies above a few GHz, where one would want the option of a using a fancier 2-loop PLL to generate the LO. (note also that the HMC1197 with PLL + VCO is 7mmx7mm & $39 from Digi-Key in singles, while HMC1097 without PLL + VCO is 4mmx4mm & $18).

4.**We can minimise routing complexity/SMP count by using clock fan-out + mux ICs**: high-quality mux/buffers like the ADCLK948 are extremely convenient and will not degrade the phase noise/stability of the DAC clock or LO.

5.**It's nice to have the option of routing the 100MHz reference directly to the HMC7044**: that way, we can use the HMC7044's integrated PLL + VCO to generate the DAC clock, without requiring a clock mezzanine or external clock source. This is a cheap, convenient option which will provide adequate performance for many applications, such as driving AOMs.

6.**I've assumed that the 100MHz reference reaching the RTM meets the phase noise spec up to >~100kHz**: if not, we'd probably want to put a 100MHz clean-up PLL on the DAC clock mezzanine, and route that to the LO mezzanine etc.

7.**The clock distribution ICs will not work for 10GHz LOs**: the fastest clock distribution ICs I'm aware of don't reach 10GHz. The ones with muxes are generally limited to <5GHz. My suggestion is that we should put mux + fan-out buffer ICs (e.g. ADCLK948) on the Sayma board to cover all LOs up to ~5GHz. Beyond this, I would suggest running coax from the LO Mezzanine to the analog mezzanine (that's about the only practical method at this frequency).

Thoughts?

[DAC clk + LO.pdf](https://github.com/m-labs/sayma/files/527897/DAC.clk.LO.pdf)

**Comment by dhslichter** [comment]
Thanks for the thoughts, Tom.  I would argue that the point of the LO generation on the Sayma (versus using an external LO source piped in on SMAs to the analog mezzanines) is to allow standalone operation/reduce clutter for low-precision applications.  No matter how careful we are, I think an onboard LO generation system will never be as high performance as one would desire for e.g. high fidelity operations on hyperfine clock qubits, and that people will probably prefer to use a phase-locked DRO or other source to provide LO to the crate for those sorts of more demanding applications.  

With that in mind, simply providing 100 MHz to the analog mezzanines and leaving it up to the mezzanine designer to implement their desired LO on the mezzanine frees us from a great deal of work and hassle on the Sayma, and allows the mezzanine designer to set things up in just their desired manner.  Remember there is no restriction on double/triple/quad-width analog mezzanines, so for example it would be possible to design a larger mezzanine card with a multi-stage ultra-low-noise PLL system for ultra-low-noise >10 GHz LO generation (if that is desired by the user), which could be distributed to multiple I/Q modulators on the analog mezzanine card itself.   If mezzanine real estate is an issue, one could reduce the channel count slightly to free room.  

When talking about clock routing, I was thinking about an application with a really simple mezzanine containing an [HMC1197](http://www.analog.com/en/products/rf-microwave/iq-modulators-demodulators/iq-modulators/hmc1197.html) or [ADF5355](http://www.analog.com/en/products/rf-microwave/pll-synth/fractional-n-plls/adf5355.html) + IQ modulator, which would need a 100 MHz reference and could then provide reasonable-quality agile microwave signals for low-precision or medium-precision applications.  

Basically what I had thought about would be sending a cleaned 100 MHz LVPECL clock (or equivalent) to each of the mezzanine boards via an additional SMP pair.  This would then let the mezzanine designer make whatever necessary design choices to achieve the required LO performance.  

@hartytp @jbqubit @gkasprow @jordens thoughts/comments?

**Comment by hartytp** [comment]
@dhslichter I agree that we should route the 100MHz analog clock to the analog mezzanines. 

You're right that one if one wants to generate the LO on the Sayma board, then the PLL could go on the (possibly quad-width) analog mezzanines(s). However, for the sake of modularity/reusability of hardware, I'd still rather have the PLL on a separate mezzanine. That way, for example, we could have two different versions of the LO PLL mezzanine -- a fancy one and a cheap one (e.g. your ADF5355) -- both of which could be used with the same (up-conversion) analog mezzanines. If we put the PLL/synth on the analog mezzanine, then we'd have to remake a lot more hardware if we ever wanted to alter the PLL.

In general, my philosophy is that I'd prefer to have more mezzanines per Sayma board, but with fewer varieties of each mezzanine to design/test/document/stock/maintain.

I do agree that we should have the option of generating the LO externally. My preferred solution for that would be to have a "passive" version of the LO mezzanine, which takes an external LO from a front-panel SMA, and distributes it to the analog mezzanines (via LVPECL buffers/multiplexers for LOs below ~5GHz, as in my diagram,  and via a passive splitter and internal SMAs for higher frequencies). This is essentially the same approach we're taking for the DAC clock. Again, this approach allows us to use the same analog mezzanine (IQ modulators + output stage) boards, regardless of whether the LO is generated internally or externally.

FWIW, having run the numbers for gate error versus phase noise, and looked at the phase noise of available microwave VOCs, I'm pretty optimistic that we can make a LO mezzanine with good enough phase noise for even our highest fidelity gates on 43Ca+ hyperfine clock qubits. Obviously, it gets a lot harder for higher frequency qubits.

**Comment by jordens** [comment]
* Why do you want a clock MUX in front of the HMC7044? The latter seems to be able to do enough clock muxing already by itself.
* I don't know whether there is enough space for a sixth mezzanine. And I would lean towards making less allowances for arbitrary LO feeding and generation setups. I would either just leave the LO generation to the ADC/DAC analog mezzanine(s) and only feed them 100 MHz or expect the sampling PLL mezzanine to do it.

**Comment by dhslichter** [comment]
I agree with @jordens here; adding an LO mezzanine, along with LO distribution/fanout, makes the Sayma cards much more complicated and requires us to overengineer at least some parts of the LO distribution system for future-proofing (which we still might screw up).  While the modularity argument is nice for motivating an LO mezzanine, I think it's better left to people designing the analog mezzanine cards, with the current front panel SMAs used for providing an external LO for simple upconverting mezzanines, which will work for most simple cases in the near term while people sort out what exactly they might want an analog mezzanine with integrated LO to do.

**Comment by hartytp** [comment]
@jordens My point was really "we should have a way of bypassing the DAC clock mezzanine and feeding the 100MHz reference directly into the HMC7044". I'm not trying to prescribe exactly how this muxing should be done, and have no objection to using a max inside the HMC7044.

**Comment by hartytp** [comment]
@jordens @dhslichter My concern with putting the LO on the analog mezzanine is that I don't want to have to re-make my carefully-designed LO PLL if I want to make a slight change to part of my input/output stage, such as a noise eater, (or, have to re-make the input/output stage if I want to alter my PLL).

> adding an LO mezzanine, along with LO distribution/fanout, makes the Sayma cards much more complicated and requires us to overengineer at least some parts of the LO distribution system for future-proofing (which we still might screw up)

I don't think the additional complexity is **that** bad: we're only talking about going from 5 to 6 mezzanines, and adding one extra fanout/mux IC for LO distribution. Still, I agree that if there isn't room for an extra LO mezzanine, then it's not an option.

> I would either just leave the LO generation to the ADC/DAC analog mezzanine(s) and only feed them 100 MHz or expect the sampling PLL mezzanine to do it.

If we don't have a separate LO mezzanine, then putting the LO on the DAC clock/sampling PLL mezzanine would be my preferred option for LOs at <~3GHz. We'd still want a mux to allow the option of feeding the 100MHz to the analog mezzanines, but we can incorporate that into the mezzanine if you'd prefer. We'd also need to make sure that the clock mezzanine is large enough to accommodate both PLLs.

**Comment by gkasprow** [comment]
Take into account that some day we may want to use RF backplane which can
not only distribute 100MHz but also 2 RF signals in GHz range. That's why I
wanted to place the LO generation in some other place.
But in such case we may always re-engineer the Sayma RTM to fit that
purpose.
In case of 100MHz mux, in order to not deteriorate the signal, we can use
just capacitor or resistor jumpers.

**Comment by hartytp** [comment]
> Take into account that some day we may want to use RF backplane which can
> not only distribute 100MHz but also 2 RF signals in GHz range. That's why I
> wanted to place the LO generation in some other place.

Okay, that makes sense. FWIW, I'm generally in favour of generating frequencies locally where conveniently possible for maximum flexibility. e.g. I can imagine a few cases where it would be desirable to have more than a single LO frequency in the same rack. 

> In case of 100MHz mux, in order to not deteriorate the signal, we can use
> just capacitor or resistor jumpers.

That could work, but it's much more convenient to have a software-programmable switch! For frequencies below ~5GHz, these are readily available with extremely low noise, that's unlikely to degrade even a high-quality LO. If we leave the muxing to the clock board, then users with LOs < 5GHz can build a clock board with an active switch, and users with higher-frequency LOs can go for a different approach.

**Comment by gkasprow** [comment]
The signal degradation caused by such switch is one thing, another is signal leaking from second, not selected input.

This can seriously  affect our reference signal.

And this is a reason where we use passive jumpers to select clock source.

If you know good LVPECL or RF swtiches with high isolation, let me know, at least we will try them. In case of troubles we can always remove serial caps.

**Comment by hartytp** [comment]
@gkasprow True, I haven't worried too much about the isolation, as low-level 100MHz leakage won't be a problem in my use case (very easy to filter off a 3GHz LO). I have a couple of ADCLK984 ICs in stock, so I can measure this if you'd like. But, you're right, if this is a problem, one can always remove the coupling caps.

Tom

**Comment by gkasprow** [comment]
Do you mean ADCLK948?

We could use it to deliver clk signal to  the clk mezzanine and also to all ADC/DAC mezzanines. In this way the bard could work without the clock mezzanine.

**Comment by hartytp** [comment]
Yes, sorry for the typo!

> We could use it to deliver clk signal to  the clk mezzanine and also to all ADC/DAC mezzanines. In this way the bard could work without the clock mezzanine.

I think that's roughly what I was suggesting, but I'd need to see a diagram/schematic to be sure I understand you correctly...

**Comment by jbqubit** [comment]
From latest RTM_DAC.step posted in #7 it looks like there is insufficient clearance between the SMA. The most compact SMA tool I've found is the following which has tip diameter 0.5". 

http://www.minicircuits.com/pdfs/HT-4-SMA.pdf

<img width="928" alt="sma_tool_tip" src="https://cloud.githubusercontent.com/assets/20543958/19439989/28700696-944e-11e6-8077-563c8e31135a.PNG">

See drawing. Can the upper row of SMA be shifted up?

**Comment by dhslichter** [comment]
Agreed here to some degree -- it seems we need a keepout with at least enough room to allow SMA nuts not to interfere with each other as they are rotated (looks in this design like it's not possible).  By spacing them out a bit more horizontally we could get more clearance.

**Comment by gkasprow** [comment]
I modified the mezzanine shape to have them distributed evenly.

**Comment by dhslichter** [comment]
What is the center-to-center spacing for adjacent SMAs in the latest design?

**Comment by dtcallcock** [comment]
How about just switching to smaller connectors?  For the analog out (upper row) it's hard to beat the SMA performance wise without going to something obscure (SSMA!?), so I would retain them.  For the analog in (lower row) connectors where we don't necessarily need the performance there are a few reasonable alternatives like MCX or MMCX (both good to 6GHz).  As these connectors sit flush with the panel they will allow much easier access to the remaining SMAs on this card and adjacent cards.

**Comment by gkasprow** [comment]
we can go for SMBs - they don't require a wrench and are less fragile than
MMCX

**Comment by dhslichter** [comment]
Crosstalk/leakage on SMB is pretty bad, and for the downconverting systems we may have inputs up at 6-10 GHz for superconducting qubit applications.  Greg, what is the center-to-center distance on the SMAs in your latest version?

**Comment by dtcallcock** [comment]
- If two ADC/DAC channels were used per input/output for IQ up/down conversion there will still be enough SMAs (assuming there is some other way of getting LO onto mezzanine).

- Use of SMBs initially doesn't preclude an all-SMA mezzanine in the future.  Particularly if the SMB+ insulating washer is made the same size as an SMA.  Also, is it actually a problem to change the panel for one with different cutouts when changing mezzanines?

- SMBs are rated to 4GHz and are larger than MCX/MMCX.  However I can see how they might be preferred from a ubiquity/ruggedness standpoint.

**Comment by gkasprow** [comment]
The centre to centre distance is 10.26mm
![smas_front](https://cloud.githubusercontent.com/assets/4325054/19494607/34ea26d0-9580-11e6-8f6d-890b6ede598d.PNG)

**Comment by dhslichter** [comment]
I think the benefits of SMA (low leakage, good high-frequency performance, good mechanical stability, many mating cycles, inexpensive connectors and cabling, high-performance cable assemblies available stock, etc) are substantial enough that we should try to stick with SMA if at all possible.  

The absolute minimum (no interference) clearance between neighboring SMAs, center to center, is 9.2 mm -- this design will give 1 mm minimum clearance between SMAs as they rotate.  When the flats are facing each other, there is 2.3 mm between flats (maximum clearance, as shown in picture).  This is enough to get one of those mini-circuits tight-clearance wrenches in, especially if you file off the pointy tips a short distance.  In any event, it is possible to fully connect and disconnect the panel with a stock wrench if one of the two neighboring SMAs is not yet connected....you can connect/disconnect everything starting at one end.  

For those groups where disconnecting and reconnecting often is in the plan, I would suggest making an external SMA patch panel elsewhere, and the cables from the RTM can be broken out there where access is easier.  

I found a different right-angle SMA part that would move the bottom row down another 0.2 mm, if that's worth the extra space: http://www.digikey.com/product-detail/en/linx-technologies-inc/CONSMA002-L/CONSMA002-L-ND/1577206

If it is possible to move the clock connector ever so slightly farther to the side, to increase its spacing with the next connector in, that would be great.

**Comment by dhslichter** [comment]
That said if it is easy to change panels we could consider going with SMB for the analog inputs, especially for the baseband mezzanine cards, where the frequencies will be lower.  Up/downconverting cards would be good to have SMA, but then we could make double-width or quad-width mezzanines with a single LO input (to be shared), which would mean fewer SMAs and thus we could space them out a bit more.

**Comment by gkasprow** [comment]
The maximum I can do is to shift the clock SMA in horizontal plane by 0.25mm

**Comment by gkasprow** [comment]
In worst case we can place some 0.5mm distance sheet between SMA THT connector and the mezzanine.
This would make assembly a little more difficult but move the SMA row down by 0.5mm.

**Comment by dhslichter** [comment]
I think we can consider this sheet spacer solution if needed -- it can be implemented late in the process since all it requires is a front panel design change.  @dtcallcock opinions?

**Comment by gkasprow** [comment]
The front panel is not designed yet, so we can implement this idea from
very beginning.


To make life easier, I can use laser-cut sheet of thick and rigid cardboard
that would be inserted during soldering and later on removed.
Anyway, during connector assembly process, we will make some tool to hold
them in right position. We can design it in such way that SMA is already
elevated.

**Comment by dhslichter** [comment]
Sounds good to me -- we should probably do some tests (S parameters/TDR) to check whether this creates a problematic impedance discontinuity to have the right-angle connector substantially offset vertically.

**Comment by gkasprow** [comment]
That's easy to check - one can compare two mezzanines with RA SMAs mounted in standard and elevated way. 
By elevating the SMA we increase the centre to centre distance to 10.56mm
Theoretically we can elevate them even more, but I'm afraid of impedance mismatch. Maybe piece of 1mm FR4 with metallized via would help :)

**Comment by sbouhabib** [comment]
I agree this has to be checked, the impedance mismatch first, then we have an unshielded small antenna (maybe just a minor issue), the idea with some board piece could be a solution, not sure about FR4 in general, I think Rogers class laminate should be for the RF parts. technically it could cause some issues for assembly, and ground continuity/repeatability of connection

**Comment by dtcallcock** [comment]
Here's a somewhat more radical idea.  I think that a dense field of ~200 SMAs (in a full rack) with sub-mm clearances is going to be basically unusable on a day-to-day basis so I can imagine that most people will take everything to patch panels.  If that's the case, then why not use a multi-coax connector on the board and supply an SMA breakout with each board?

There are quite a few solutions out there such as the [Smith MDHC](http://www.smithsconnectors.com/us/products/product-catalog/mdcx-and-mdhc-coax).  Other suppliers include [Times](http://www.timesmicrowave.com/cms/products/specialized-products-gallery/high-density-interconnects/high-density-interconnect-overview/index.shtml), [Carlisle](http://www.carlisleit.com/sites/default/files/resources/documents/HDRFI_Brochure_0.pdf) and [H&S.](http://www.hubersuhner.com/en/Products/Radio-Frequency/Connectors).  I haven't dug into specs too much but it seems you can get beyond-SMA electrical performance.

Whilst I think this will make a much nicer final product, it will presumably add a lot of cost, increases design complexity and is more likely to be obsoleted than SMA (still going strong at 50+).

**Comment by jmizrahi** [comment]
@dtcallcock in general I think it's a good idea to avoid obscure, expensive connectors and cables if at all possible. I also don't really see what your solution solves -- it requires an SMA breakout panel, just as before. It's definitely more elegant, but I don't think that's a strong enough argument to move away from SMA.

I would say that so long as it is possible to connect the cables with a wrench, even if it has to be done in sequential order, then it's perfectly usable and I would stick with SMA. You wouldn't even necessarily need a patch panel -- a bunch of 100mm SMA cables could just be permanently attached to the front panel, and then all connections and disconnections happen with the 100mm cables.

**Comment by allcockd** [comment]
>I also don't really see what your solution solves -- it requires an SMA breakout panel

Probably several panels.  These can then have adequate connector spacing, different connectors depending on requirements, and space for sensible cable management.

>avoid obscure, expensive connectors

Probably no more obscure than the other connector on the mezzanine.  As for cost, I'll admit I have no idea (though decent SMAs aren't exactly cheap and you have to but the cable either way).  Anyone have experience here @sbouhabib?

>even if it has to be done in sequential order, then it's perfectly usable 

Sure, for one or two cards this is fine but my worry is how well it scales to ~200 connections.  It's going to be a pig to change cards, debug damaged cables/connectors and could easily be a rats' nest that puts a lot of mechanical force on the cards and connectors.  Also a bundle of 100mm cables and i-pieces doesn't seem the best solution for crosstalk, ground loops and maintaining the phase and amplitude stability of microwave signals.  I'm sure each lab will come up with a reasonable coping strategy but I thought this might be a good juncture to engineer a nice solution.  

It seems like the consensus is that I'm overstating the problem though so I'm happy to drop this.

**Comment by dhslichter** [comment]
Two different github accounts @dtcallcock @allcockd ?  Fancy pants!  

The other mezzanine connectors are actually standard SMP (~$6 each), plus the power/io pin connector (also cheap).  The spec'ed SMA connectors are ~$20-$25 each.  My guess about these sorts of integrated connectors, based on what gets charged for standard Micro-D and other fancy space-hardware stuff, is that they'd be several hundred dollars apiece at a minimum.  Factor in that you are buying a mating pair as well, with one end integrated into a cable assembly that likely can only be manufactured by the company that makes the fancy connector.  $$$$$$$ :D

That said, I don't think we should be ready to drop this right away.  We do need to consider that either way (SMA or multi-connector) one is going to want a sturdily mounted SMA breakout panel, and this panel can be used for routing externally generated LO and mounting the required components, among other tasks.  What happens to the godawful ratsnest of cables that would emerge from a fully populated crate is going to have to get figured out by each lab independently -- connector changes have no effect there one way or the other.

**Comment by dhslichter** [comment]
Also would need the shields of the coax cables not to be grounded together or to the panel if we care about ground loops a la @hartytp

**Comment by dtcallcock** [comment]
> plus the power/io pin connector (also cheap)

I mean isn't this connector an obscure Samtec interface (not that there's anything wrong with that)?

>$$$$$$$

But I'd be interested to know actual cost/feasibility from someone who's done something like this.

> connector changes have no effect there one way or the other.

The ratsnest can be a lot tidier if it's a smaller number of blind mate connectors and wiring harnesses and not a bunch of cables you have to be able to delve into with your hand and an SMA wrench.

>What happens to the godawful ratsnest of cables that would emerge from a fully populated crate is going to have to get figured out by each lab independently

Even if we stick with SMAs it'd be good to share ideas on how to do this.  Perhaps something to figure out once we've got prototypes and tried a few things out.

I have no idea who @allcockd is.

**Comment by jmizrahi** [comment]
@dtcallcock my thinking was along the lines of what @dhslichter said, namely that these connectors would be rather custom and quite expensive, and likely unique to whatever company they came from (that's what I meant by "obscure"). But, you make a strong argument that it would make the front panel quite a bit more manageable. Would you be willing to get some quotes so that we have concrete numbers to discuss? I also hadn't realized just how expensive those SMA connectors are -- at $25 a piece, then a $200 8x connector would be a wash, at least as far as direct connector cost is concerned.

**Comment by sbourdeauducq** [comment]
I also like Smith MDHC. Quote requested.

**Comment by gkasprow** [comment]
Samer, you shown us during last meeting at WUT some DSUB-like multipin RF connectors.

Do you remember their PN?

**Comment by gkasprow** [comment]
These are the fbm connectors.but after 2ghz they become quite "hard" to
match properly.theoretically doable but hard and in general  narrowbanded

Samer Bou Habib

20.10.2016 10:56 "Grzegorz Kasprowicz" <G.Kasprowicz@elka.pw.edu.pl>
napisał(a):

>
>
> Samer, you shown us during last meeting at WUT some DSUB-like multipin RF
> connectors.
>
> Do you remember their PN?
>

**Comment by dhslichter** [comment]
It looks like a 4x connector (single-width mezzanine) is 35.6 mm wide, so I don't know if we can really put them all next to each other (mezzanines are 32 mm wide now, not sure what the gap is). 

Thus if we go Smith MDHC, we should use double-width mezzanine cards with 8x connectors (53.4 mm width, easily fits).  Thus there would be 2 connectors per RTM, covering all 16 DAC/ADC lines.

**Comment by dhslichter** [comment]
MDHC has --85 to -95 dB crosstalk between connectors, return loss better than -20 dB to 13 GHz....I'd say performance-wise it's definitely suitable.

**Comment by dtcallcock** [comment]
I've also asked for quotes on:

[Carlisle CoreHC](http://www.carlisleit.com/sites/default/files/resources/documents/Final_CoreHC_1.4.16_0.pdf)
[Carlisle Core GD](http://www.carlisleit.com/sites/default/files/CoreGD_0.pdf)
[Rosenberger Multiport Minicoax](https://www.rosenberger.com/0_documents/de/catalogs/ba_tm/TM_Multiport_Flyer_2016.pdf)
The latter system has a few parts on Mouser to give you an idea: [link](http://www.mouser.com/ProductDetail/Rosenberger/23C21F-40ML5/?qs=sGAEpiMZZMuLQf%252bEuFsOriDPIEqVsEWKy%2fFTfPHzMKUj2LkM4Qwpnw%3d%3d)

These might be a bit too fine pitch and fragile but see what you think.

**Comment by dhslichter** [comment]
Any love on these quotes @dtcallcock @sbourdeauducq ?  What pricing/lead times are being given?

**Comment by sbourdeauducq** [comment]
No reply. BTW we should break down this issue into smaller ones.

**Comment by dtcallcock** [comment]
I got the Rosenberger quote.  Still working on Carlisle.

Rosenberger 23C21E-40ML5
4ch solder pin right angle
Qty 100-250 $56.82ea, 5-7wk lead time
[Datasheet](http://rosenberger.de/ok/images/documents/db/23C21E-40ML5.pdf)

Rosenberger L99-816-30E
Cable set 4ch, 30cm, SMA M termination
Qty 100-250 $157.01, 13-15wk lead time
[Datasheet](http://rosenberger.de/ok/images/documents/db/L99-816-XXE_DB.pdf)

As I said, these things might be _too_ compact.

**Comment by jbqubit** [comment]
@dtcallcock Thank you for getting these quotes. Price is good economy vs individual SMA. 
@sbourdeauducq Please break this issue into smaller ones.

**Comment by dhslichter** [comment]
@jbqubit @sbourdeauducq I will break this down into smaller issues.

**Comment by gkasprow** [comment]
http://uk.rs-online.com/web/p/sma-connectors/7123264/
http://uk.rs-online.com/web/p/sma-connectors/7123254/
